### PR TITLE
Issue #SB-7316 fix:UI for displaying contribution list and dropdown of users and organization.

### PR DESCRIPTION
**Pull request by rajeshcompass**
**Description:**
Should display content creators and organizations list when creating new content and added resources like collection. course and lessons.and should also show the ownership field in edit form and display organization and user name in drop down on the basis of configuration value.

**Testcases and Scenarios:**
https://project-sunbird.atlassian.net/browse/SB-7241

### CODE REVIEW

File: org.ekstep.metadata-1.1/editor/controller.js [hunk 1]
```diff
@@ -78,7 +78,21 @@ angular.module('org.ekstep.metadataform', []).controller('metadataForm', ['$scop
...
+        "inputType": "select",
+        "label": "Owner",
+        "range": {},
+        "name": "level",
+        "placeholder": "Owner",
+        "renderingHints": {},
+        "required": true,
+        "visible": true
+    };
+    $scope.selectedOwnerId;
```
Comment (pallakartheekreddy) [inline org.ekstep.metadata-1.1/editor/controller.js]: What is the use of this variable?

File: org.ekstep.sunbirdcommonheader-1.5/editor/headerApp.js [hunk 2]
```diff
@@ -59,6 +58,50 @@ angular.module('org.ekstep.sunbirdcommonheader:app', ["Scope.safeApply", "yaru22
...
+                'type': node.data.metadata.owershipType === 'createdFor' ? 'organisation' : 'user'});
+                $scope.contentCredits = ecEditor._.uniqBy($scope.contentCredits, "id");
+            }
+        });
+    }
+
+    /*
+    * Initialize listLimit value when click on user icon button.
+    */
+    $scope.showContributionList = function() {
```
Comment (pallakartheekreddy) [inline org.ekstep.sunbirdcommonheader-1.5/editor/headerApp.js]: Why you added a function to store listLimit value?

File: org.ekstep.sunbirdcommonheader-1.5/editor/headerTemplate.html [hunk 3]
```diff
@@ -93,4 +100,5 @@
 </div>
 <script>
     $(".ui.dropdown").dropdown();
+    $("#dropdownitem .item").addClass('disabled');
```
Comment (pallakartheekreddy) [inline org.ekstep.sunbirdcommonheader-1.5/editor/headerTemplate.html]: don't use `dropdownitem` as id name. Use ContentCredits/ Credits

File: org.ekstep.metadata-1.1/editor/controller.js [hunk 4]
```diff
@@ -419,9 +445,35 @@ angular.module('org.ekstep.metadataform', []).controller('metadataForm', ['$scop
...
         !EventBus.hasEventListener('metadata:form:getdata') && ecEditor.addEventListener('metadata:form:getdata', $scope.getScopeMeta, $scope);
         var callbackFn = function(config) {
+            config.fields = config.fields.filter(function( obj ) {
+                return obj.code !== 'ownedBy';
+            });
             $scope.fields = config.fields;
             $scope.messages = config.messages || {};
             $scope.tempalteName = config.template;
+
+            if($scope.tempalteName  === 'defaultTemplate') {
```
Comment (vinukumar-vs) [inline org.ekstep.metadata-1.1/editor/controller.js]: This code should be simplified.

File: org.ekstep.sunbirdcommonheader-1.5/editor/headerTemplate.html [hunk 5]
```diff
@@ -23,7 +23,18 @@
         <div class="five wide column custom-5-wide" style="padding-top:1rem!important;">
             <div class="row header-icon-container">
                 <div class="custom-row-2">
+                <div class="content-contributors">
+                    <p>Created with contribution from <span style="color: blue">{{contentCredits.length}}</span></p>
+                </div>
                     <div class="icon-group-2">
+                        <div class="ownerlist-dropdown labeled search icon" id="dropdownitem">
```
Comment (vinukumar-vs) [inline org.ekstep.sunbirdcommonheader-1.5/editor/headerTemplate.html]: id="dropdownitem" id name should be specific name. ownerListDP

File: org.ekstep.sunbirdcommonheader-1.5/editor/headerApp.js [hunk 6]
```diff
@@ -59,6 +58,56 @@ angular.module('org.ekstep.sunbirdcommonheader:app', ["Scope.safeApply", "yaru22
...
+    /*
+    * Add owner details and update current count with new values.
+    */
+    $scope.updateOwnershipList = function(event, nodeData) {
+        var contentCredit = {'ownedBy':nodeData.data.metadata.identifier, 
+                                'owner':nodeData.data.metadata.creator, 
+                                'ownershipType': nodeData.data.metadata.contentType };
+        
+        var index = _.findIndex($scope.contentCredits, function(o) {
+                            return o.owner == nodeData.data.metadata.creator; });
```
Comment (vinukumar-vs) [inline org.ekstep.sunbirdcommonheader-1.5/editor/headerApp.js]: nodeData.data.metadata.creator replace this with contentCredit.owner

File: org.ekstep.sunbirdcommonheader-1.5/editor/headerApp.js [hunk 7]
```diff
@@ -59,6 +58,56 @@ angular.module('org.ekstep.sunbirdcommonheader:app', ["Scope.safeApply", "yaru22
...
+        if(nodeData.data.metadata.creator && index == -1) {
```
Comment (vinukumar-vs) [inline org.ekstep.sunbirdcommonheader-1.5/editor/headerApp.js]: nodeData.data.metadata.creator replace with contentCredit.owner

File: org.ekstep.sunbirdcommonheader-1.5/editor/headerApp.js [hunk 8]
```diff
@@ -59,6 +58,56 @@ angular.module('org.ekstep.sunbirdcommonheader:app', ["Scope.safeApply", "yaru22
     $scope.isFlagReviewer = false;
     $scope.editorEnv = "";
     $scope.showEditMeta = true;
+    $scope.contentCredits = [];
+    $scope.listLimit = 5;
+    /*
+    * Add owner details and update current count with new values.
+    */
+    $scope.updateOwnershipList = function(event, nodeData) {
+        var contentCredit = {'ownedBy':nodeData.data.metadata.identifier,
```
Comment (vinukumar-vs) [inline org.ekstep.sunbirdcommonheader-1.5/editor/headerApp.js]: Add one more key "resourcesAdded": to this object. Use this key while removing/deleting the resource. 
If resourcesAdded = 1, then you can remove the user from the list directly.
If resourcesAdded = 2, then you can should remove the user from the list, you just have to decrement the resourceAdded count.

File: org.ekstep.sunbirdcommonheader-1.5/editor/headerApp.js [hunk 9]
```diff
@@ -59,6 +58,56 @@ angular.module('org.ekstep.sunbirdcommonheader:app', ["Scope.safeApply", "yaru22
     $scope.isFlagReviewer = false;
     $scope.editorEnv = "";
     $scope.showEditMeta = true;
+    $scope.contentCredits = [];
+    $scope.listLimit = 5;
+    /*
+    * Add owner details and update current count with new values.
+    */
+    $scope.updateOwnershipList = function(event, nodeData) {
```
Comment (vinukumar-vs) [inline org.ekstep.sunbirdcommonheader-1.5/editor/headerApp.js]: Where is the logic while editing the couser, textook getting all the Owsership list? 

Is this function is calling while editing as well(node.added event should trigger)?

File: org.ekstep.sunbirdcommonheader-1.5/editor/headerApp.js [hunk 10]
```diff
@@ -59,6 +58,56 @@ angular.module('org.ekstep.sunbirdcommonheader:app', ["Scope.safeApply", "yaru22
...
+    * Initialize listLimit value when click on user icon button.
+    */
+    $scope.showContributionList = function() {
+        $scope.listLimit = 5;
+    };
+
+    /*
+    * Increase size of listLimit when click on see more button.
+    */
+    $scope.addListSize = function() {
```
Comment (vinukumar-vs) [inline org.ekstep.sunbirdcommonheader-1.5/editor/headerApp.js]: Why are you showing only +5 when we click on showmore?

Show entire list with scrollbar when it is more than 10.

File: org.ekstep.sunbirdcommonheader-1.5/editor/headerApp.js [hunk 11]
```diff
@@ -59,6 +58,56 @@ angular.module('org.ekstep.sunbirdcommonheader:app', ["Scope.safeApply", "yaru22
...
+            if(node.data.metadata.creator && index == -1) {
+                $scope.contentCredits.push(contentCredit);
+            }
+        });
+    }
+
+    /*
+    * Initialize listLimit value when click on user icon button.
+    */
+    $scope.showContributionList = function() {
```
Comment (vinukumar-vs) [inline org.ekstep.sunbirdcommonheader-1.5/editor/headerApp.js]: Change the function name resetList.

File: org.ekstep.sunbirdcommonheader-1.5/editor/headerApp.js (hunk 8 above)
Comment (rajeshcompass) [reply org.ekstep.sunbirdcommonheader-1.5/editor/headerApp.js]: This logic will help only if removing single resource at a time but removing a child and that child have another child then this logic will not help.

**Pull request merged**